woodwork/logical_types.py

```diff
@@ -306,10 +306,39 @@ def _remove_timezone(self, series):
 
     def transform(self, series, null_invalid_values=False):
         """Converts the series data to a formatted datetime. Datetime format will be inferred if datetime_format is None."""
+
+        def _year_filter(date):
+            """Applies a filter to the years to ensure that the pivot point isn't too far forward."""
+            if date.year > 2030:
+                date = date.replace(year=date.year - 100)
+            return date
+
+        apply_filter = False
         new_dtype = self._get_valid_dtype(type(series))
         series = self._remove_timezone(series)
         series_dtype = str(series.dtype)
 
+        if series_dtype in ["object", "string"]:
+            numbers_in_date = []
+            for (
+                _,
+                series_val,
+            ) in (
+                series.iteritems()
+            ):  # use iteritems to support iterating through pyspark series
+                try:
+                    vals = re.findall(r"\d+", series_val)
+                    if len(vals):
+                        numbers_in_date.append(vals)
+                except TypeError:
+                    # for values like NaT, None, etc that aren't string/byte types
+                    continue
+            list_lengths = list(
+                map(lambda x: max([len(g) for g in x]), numbers_in_date),
+            )
+            if len(list_lengths) and max(list_lengths) == 2:
```

Comment:
this is definitely reliable right? There's no other > 2 digit numbers we have to worry about?

Reply:
Yep, this is just checking that the longest value in our provided datetime is 2 digits, and the filter only applies to year values. This is reliable for changing the year value if it appears since it guarantees that the passed in year value has only 2 digits

Reply:
Yeah - I'm just wondering if there's not a date that looks like 10/23/22 1230PM or something like that.